vayesta/solver/coupling.py

```diff
@@ -354,22 +525,28 @@ def _get_delta_t_for_delta_tailor(fragment, fock):
     return dt1, dt2
 
 
-def externally_correct(solver, external_corrections):
+def externally_correct(solver, external_corrections, eris=None):
     """Build callback function for CCSD, to add external correction from other fragments.
 
     TODO: combine with `tailor_with_fragments`?
 
     Parameters
     ----------
-    solver: CCSD_Solver
+    solver : CCSD_Solver
         Vayesta CCSD solver.
-    external_corrections: list[tuple(int, str, int)]
+    external_corrections : list of tuple of (int, str, int)
         List of external corrections. Each tuple contains the fragment ID, type of correction,
         and number of projectors for the given external correction.
+    eris : _ChemistsERIs
+        ERIs for parent CCSD fragment. Used for MO energies in residual contraction, and for
+        type of correction == 'external-ccsdv', where the parent Coulomb integral is contracted.
+        If not passed in, MO energy if needed will be constructed from the diagonal of 
+        get_fock() of embedding base class, and the eris will be also be obtained from the
+        embedding base class. Optional.
```

Comment:
Why do we not always just use the embedding class attributes?

Reply:
Not sure what you mean here

Reply:
So the eri's themselves are potentially used here too, when we are contracting with the low-level coulomb integrals. I think for solids, there may be differences in MO energies regarding whether they are the shifted ones or not?